src/game/savegame_bson.c

```diff
@@ -147,6 +149,8 @@ static struct json_value_s *Savegame_BSON_ParseFromBuffer(
         return NULL;
     }
 
+    g_GameInfo.save_initial_version = header->initial_version;
```

Comment:
This does not belong in here. This function is stateless and is used to parse a BSON savegame file, and this patch introduces state to it by making the function interact with global variables. I admit we do it all over the place, but this example is particularly bad, especially considering that later you try to cover for this side effect by the introduction of `backup_save_slot` in the scanner.

Instead, this assignment should be only executed when the user chooses to load the game, for example like this:

```diff
diff --git a/src/game/savegame.c b/src/game/savegame.c
index c1cb1784..3ef0a005 100644
--- a/src/game/savegame.c
+++ b/src/game/savegame.c
@@ -342,6 +342,8 @@ bool Savegame_Load(int32_t slot_num, GAME_INFO *game_info)
         Savegame_LoadPostprocess();
     }
 
+    g_GameInfo.save_initial_version = m_SavegameInfo[slot_num].initial_version;
+
     return ret;
 }
 
@@ -445,6 +447,7 @@ bool Savegame_LoadOnlyResumeInfo(int32_t slot_num, GAME_INFO *game_info)
         strategy++;
     }
 
+    g_GameInfo.save_initial_version = m_SavegameInfo[slot_num].initial_version;
     return ret;
 }
 
@@ -464,7 +467,6 @@ void Savegame_ScanSavedGames(void)
 {
     Savegame_Shutdown();
 
-    uint16_t backup_initial_version = g_GameInfo.save_initial_version;
     g_SaveCounter = 0;
     g_SavedGamesCount = 0;
 
@@ -558,8 +560,6 @@ void Savegame_ScanSavedGames(void)
     }
 
     g_SaveCounter++;
-
-    g_GameInfo.save_initial_version = backup_initial_version;
 }
 
 void Savegame_ScanAvailableLevels(REQUEST_INFO *req)
diff --git a/src/game/savegame_bson.c b/src/game/savegame_bson.c
index bcd1bd7e..50fd35e0 100644
--- a/src/game/savegame_bson.c
+++ b/src/game/savegame_bson.c
@@ -149,8 +149,6 @@ static struct json_value_s *Savegame_BSON_ParseFromBuffer(
         return NULL;
     }
 
-    g_GameInfo.save_initial_version = header->initial_version;
-
     struct json_value_s *root = bson_parse(uncompressed, uncompressed_size);
     Memory_FreePointer(&uncompressed);
     return root;
```

Reply:
>I put it in the scanner because the select level passport menu needs to know the initial version

In case your concern here is that `SAVEGAME_INFO.initial_version` is not available for public, we can create a getter for it: `SaveGame_GetInitialVersion(int32_t slot_num)`

Reply:
For the first point, the issue was solved by adding in initial_version to the fillinfo function. I think it looks good.

For the second, it should be ok right? `int16_t` will still be able to store thousands of more versions than we need. I think it will be useful to have the TombATI version defined. I'm going to add it.